pyuvdata/tests/test_uvcal.py

```diff
@@ -138,105 +135,110 @@ def teardown(self):
 
     def test_equality(self):
         """Basic equality test"""
-        nt.assert_equal(self.gain_object, self.gain_object)
+        assert self.gain_object == self.gain_object
 
     def test_check(self):
         """Test that parameter checks run properly"""
-        nt.assert_true(self.gain_object.check())
+        assert self.gain_object.check()
 
     def test_nants_data_telescope(self):
-        self.gain_object.Nants_data = self.gain_object.Nants_telescope - 1
-        nt.assert_true(self.gain_object.check)
+        # make sure it's okay for Nants_telescope to be strictly greater than Nants_data
+        self.gain_object.Nants_telescope += 1
+        # add dummy information for "new antenna" to pass object check
+        self.gain_object.antenna_names = np.concatenate((self.gain_object.antenna_names, ["dummy_ant"]))
+        self.gain_object.antenna_numbers = np.concatenate((self.gain_object.antenna_numbers, [20]))
+        assert self.gain_object.check()
+        # make sure an error is raised if Nants_data is strictly greater than Nants_telescope
         self.gain_object.Nants_data = self.gain_object.Nants_telescope + 1
```

Comment:
I'm concerned that this might be raising the wrong error. Changing Nants_data could cause errors because Nants_data is just wrong, not because it's smaller than Nants_telescope.

Reply:
I think this part of the test existed even before the additions @plaplant made :thinking: I think it may just be throwing a ValueError because some attributes have shapes `(Nants_data, blah)`. Is it worth trying to rework this test, or just changing the caption to be explicit about the fact we're basically testing `check`?

Reply:
I just inferred the reason for the test from its general structure. If indeed the test is raising an error because things like the `gain_array` are the wrong size, we should modify dimensions of data in the test so that the _only_ reason for failing the test is the incompatibility between `Nants_data` and `Nants_telescope`.

Reply:
yep, I just checked and it's not raising the advertised error (another reason to move toward always specifying the expected error message). I'll push a fix soon.

Reply:
Interestingly, when I tried to fix it it didn't error! So I added something to the check method to catch this problem, both in uvcal and in uvdata (and added tests for them as well)